app/models/subsidiary.rb

```diff
@@ -1,17 +1,26 @@
 class Subsidiary
-  attr_reader :id, :name, :address, :cep
+  attr_reader :id, :name, :address, :cnpj, :token
 
-  def initialize(id:, name:, address:, cep:)
+  def initialize(id:, name:, address:, cnpj:, token:)
     @id = id
     @name = name
     @address = address
-    @cep = cep
+    @cnpj = cnpj
+    @token = token
   end
 
   def self.all
-    [new(id: 1, name: 'Vila Maria', address: 'Avenida Osvaldo Reis, 801', cep: '88306-773'),
-     new(id: 2, name: 'Ipiranga', address: 'Rua da Concórdia, 201', cep: '57071-812'),
-     new(id: 3, name: 'Santos', address: 'Rua das Hortências, 302', cep: '78150-384')]
+    response = Faraday.get "#{Rails.configuration.apis['subsidiaries']}/subsidiaries"
+
+    if response.status == 200
+      list = JSON.parse(response.body, symbolize_names: true)
+      list.map do |item|
+        new(id: item[:id], name: item[:name],
+            address: item[:address], cnpj: item[:cnpj], token: item[:token])
```

Comment:
```suggestion
        new(item)
```

Acho que como item é um hash talvez dê pra fazer assim direto 😄

Reply:
meu arquivo de api get_subsidiaries.json estava assim:
{
    "id":1,
    "name":"Lorem1",
    "address":"R. Lorem Ipsum, XXX, Lorem, Ipsum",
    "created_at":"2020-10-07T15:38:01.558Z",
    "updated_at":"2020-10-07T15:38:01.558Z",
    "cnpj":"73.590.707/9101-24",
    "token":"9F7IYV"
  }
quando montava dessa forma "new(item)" dava problema pq tinha esses dois argumentos "created_at" e "updated_at" que não estão no initialize, então vou remove-los do json pq ai ele monta o new direitinho.

Reply:
meu arquivo de api get_subsidiaries.json estava assim:
{
"id":1,
"name":"Lorem1",
"address":"R. Lorem Ipsum, XXX, Lorem, Ipsum",
"created_at":"2020-10-07T15:38:01.558Z",
"updated_at":"2020-10-07T15:38:01.558Z",
"cnpj":"73.590.707/9101-24",
"token":"9F7IYV"
}
Quando uso new(item) da problema por causa desses dois paramentros,"created_at"  e "updated_at", que nao estao no initialize mas se eu remove-los do json ai ele monta o new(item) direitinho.